### DISCUSSION TITLE
Links to posts do not work on my site

### DISCUSSION THREAD

**Discussion by dev-hanzo** [ObsidianPublisher/obsidian-github-publisher#67] [Q&A — Plugin] [answered]
- Command used : 
	- [ ] SHARE ON FILE
	- [x] SHARE ALL SHARED FILE
	- [ ] Update the repository with new file only
	- [ ] Update the repository with edited file only
	- [ ] Update the repository with edited and new file
	- [ ] Clean the repository

- **My plugin data** : 
[data - Copy.txt](https://github.com/ObsidianPublisher/obsidian-github-publisher/files/10369083/data.-.Copy.txt)

- **The file I to send** 
[new post.md.txt](https://github.com/ObsidianPublisher/obsidian-github-publisher/files/10369143/new.post.md.txt)
[post-1.md.txt](https://github.com/ObsidianPublisher/obsidian-github-publisher/files/10369119/post-1.md.txt)

- **Screenshot to the log** 
![image](https://user-images.githubusercontent.com/117753674/211212104-b930dfa9-9fb8-4d0e-84c0-b4af9de79f2f.png)

**location of files on my drive**
```
.
├─ INBOX/
│  └─ wiki/
│  │   └─ _Files_/
│  │   │   └─ pic-1.svg 
│  │   ├─ new post.md      (*link-2)
│  │   └─ last post.md  
│  └─ post-1.md            (*link-1)
└─ .obsidian

```
*link-1
https://lovely-blini-8cc43b.netlify.app/first%20post/

*link-2
https://lovely-blini-8cc43b.netlify.app/wiki/post-g2/

**my problems:**
1. Links to posts do not work on my site
2. I am unable to link between these two files
3. sometimes links to images don't work either
4. not all file names work to replace "spaces" with "-"

**alleged cause**
if a file is assigned a category, then the category name will be added to all links inside this file

**my goal**
I want to make links to files in notes that work on the site and on the github

**Comment by Mara-Li** [comment, marked as answer]
I will check. The alleged cause seems to be false.

Can i get the raw converted file?

Normally, all links are relative to the file.
Also, can I get the build log?

**Reply by Mara-Li** [reply]
(BTW I will try to repro this problem on my side after eating!)

**Reply by dev-hanzo** [reply]
yes, send log my bild

[bild.txt](https://github.com/ObsidianPublisher/obsidian-github-publisher/files/10369206/bild.txt)

**Reply by dev-hanzo** [reply]
~~~
  | <p><a href="../first post">post-1</a></p>
  | <p><img alt="" src="../../_Files_/pic-1.svg"/></p>
  | <p><img alt="pic200" src="../../_Files_/pic-1.svg"/></p>
  | <script>


<h1>post G2</h1>
<p><a href="[../first post](https://lovely-blini-8cc43b.netlify.app/wiki/first%20post)">post-1</a></p>
<p><img alt="" src="[../../_Files_/pic-1.svg](https://lovely-blini-8cc43b.netlify.app/_Files_/pic-1.svg)"/></p>
<p><img alt="pic200" src="[../../_Files_/pic-1.svg](https://lovely-blini-8cc43b.netlify.app/_Files_/pic-1.svg)"/></p>
~~~


https://lovely-blini-8cc43b.netlify.app/  **wiki**/first%20post  )

**Reply by Mara-Li** [reply]
No ! I need the file on your github repo, not the HTML file :)

**Comment by Mara-Li** [comment]
Okay, it seems the name are not converted : 
```
10:41:58 PM: WARNING  -  Documentation file 'first post.md' contains a link to 'new%20post.md' which is not found in the documentation files.
```
Normally, in conversion, New post must be renamed.

**Comment by Mara-Li** [comment]
Okay, alors, I noticed some error : 
1. In `new post.md` : 
	- Like to the image is wrong : You need to link to `Files` not `_Files_`
	- There are an error in the frontmatter, maybe it's the cause of the bug. 
	![image](https://user-images.githubusercontent.com/30244939/211217204-fd455635-b0c4-4bf0-87d4-85067c9e8e11.png)
2. It seems that your blog is not well configured, notability for the icon. Did you change everything to point to your blog ? 


BTW, 
I found the problem. It seems that markdown links are not converted (or not well). Wikilinks work perfectly.

I will fix that in the night, thanks.

**Reply by dev-hanzo** [reply]
>_You need to link to Files not _Files__

i need use this link formats in obsidian (both options)
1. Relative path to file
2. Absolute path in vault

**Reply by Mara-Li** [reply]
It's not relevant here

**Comment by Mara-Li** [comment]
Error fixed in last version btw

**Reply by dev-hanzo** [reply]
thank you very much. i will check after two hour

**Reply by dev-hanzo** [reply]
one link good, second link bad 😞
https://lovely-blini-8cc43b.netlify.app/wiki/post-g2/
https://github.com/peter-hanzo/pub-0/blob/main/docs/wiki/post-G2.md

![image](https://user-images.githubusercontent.com/117753674/211257686-03094711-e596-47d0-b2a9-83b7cd94a1ef.png)

![image](https://user-images.githubusercontent.com/117753674/211264269-28d331bd-6502-4020-aa84-bd2d6288d884.png)

I made it hard two file to show possible errors
all this links choices work in obsidian
please see and use them 🤗

[new post.md.txt](https://github.com/ObsidianPublisher/obsidian-github-publisher/files/10371468/new.post.md.txt)
[post-1.md.txt](https://github.com/ObsidianPublisher/obsidian-github-publisher/files/10371469/post-1.md.txt)

**Comment by Mara-Li** [comment]
Seems to be a limitation with regex. Could you try with only one version of each try ?

**Comment by Mara-Li** [comment]
![image](https://user-images.githubusercontent.com/30244939/211266552-4b650784-39b5-43bf-abdc-0925c830bef5.png)
Seems to be error of links here

**Reply by Mara-Li** [reply]
I think the best way to prevent links error is to use the shortest past possible in Obsidian

**Reply by dev-hanzo** [reply]
yes I am starting to think too
but sometimes I need Absolute path in vault

**Reply by dev-hanzo** [reply]
> Seems to be error of links here

I do not have this error

**Reply by dev-hanzo** [reply]
> Could you try with only one version of each try ?

how can I this do? please tell me my steps

**Comment by dev-hanzo** [comment]
I changed name of one of files and almost all working 

https://lovely-blini-8cc43b.netlify.app/wiki/post-g2/

but did not work replace "spaces" with "-"
and not work the links in block

**Reply by dev-hanzo** [reply]
**the errors in github**


![image](https://user-images.githubusercontent.com/117753674/211277649-a3da8e46-a77c-4591-9c3d-3a1483f57931.png)

**Comment by Mara-Li** [comment]
There is too much error in my code T_T

Thank you for your help to find them ! 

I fixed some, i need to found another bug I spotted : 

- If the link not exists, obviously, they won't be convert ; You need to check the link exist  in obsidian
- If, in markdown link, the alt text is the same of the basename (of file), the converted link will be reversed (`[./first post.md](post1)` instead of `[post 1](./first%20post.md)`)

I have another bug, and it's pretty strange, dunno if it's a bug on my side or github side, because I can't found anything in my code T_T

Also, can you please send me the tree of your vault ? 

So I can debug in better condition!

**Reply by Mara-Li** [reply]
Found another things : It seems that github doesn't like link to # ? Dunno why tbh.

**Comment by Mara-Li** [comment]
Okay, I think i found every error on my side ? Could  you please check if every is fine in the new version ? (4.11.5)

EDIT: Forgot to edit somethings

**Comment by dev-hanzo** [comment]
**the tree of my vault**
.
├─ INBOX/
|  └─ _Files_/
│  └─ My folder/
│  │   └─ Files-2/
│  │   │   └─ pic-1.svg 
│  │   ├─ new post.md      (https://lovely-blini-8cc43b.netlify.app/wiki/post-g2/)
│  │   └─ last post.md  
│  └─ post 1.md            (https://lovely-blini-8cc43b.netlify.app/first%20my%20post/)
└─ .obsidian

I check in next couple of hours new version

**Comment by dev-hanzo** [comment]
![image](https://user-images.githubusercontent.com/117753674/211297418-8e9ab0be-79a5-47d6-9223-211fedf5e563.png)

so, I checked new version

**1. in githab all right, all links work
2.  in site once of two links work
3.  the problem change to replace "spaces" with "-" remains** 



![image](https://user-images.githubusercontent.com/117753674/211310336-655e849e-1597-48fc-9641-a0340672e388.png)

**Reply by dev-hanzo** [reply]
"Relative path to file" and "Absolute path in vault" 
**it is working**

**Reply by Mara-Li** [reply]
Hmm, for me the title replacement work as intented ; space are replaced by `-`

**Reply by dev-hanzo** [reply]
one files name of two did replaced

![image](https://user-images.githubusercontent.com/117753674/211324363-64615748-a8f5-4894-a83a-a38c72422698.png)

**Comment by Mara-Li** [comment]
In my side, everything seems to work as attended, except for the plugin that embed files. I need to check it.
![image](https://user-images.githubusercontent.com/30244939/211323879-e053a5ff-50ae-4977-9f0b-423ace9f8094.png)

**Comment by dev-hanzo** [comment]
https://lovely-blini-8cc43b.netlify.app/first%20my%20post  **.md** <---- bad link
https://lovely-blini-8cc43b.netlify.app/wiki/post-g2/    <---- good link

**Comment by Mara-Li** [comment]
I really can't repro :/
What do you have as regex for that ?

**Reply by dev-hanzo** [reply]
i propose decided a problem links in site
replaced it is little  problem

no, i do not use regex

**Comment by dev-hanzo** [comment]
so, i will try again by new files

**Comment by dev-hanzo** [comment]
**files in my disk**
```
├─ INBOX/
|  └─ _Files_/
│  └─ My folder/
│  │   └─ Files2/
│  │   │   └─ pic-1.svg 
│  │   ├─ new2 post.md     
│  │   └─ last post.md  
│  └─ post 4.md           
└─ .obsidian
```
**data file**
[data - Copy.json.txt](https://github.com/ObsidianPublisher/obsidian-github-publisher/files/10374186/data.-.Copy.json.txt)

**sent files**
[new2 post.md.txt](https://github.com/ObsidianPublisher/obsidian-github-publisher/files/10374189/new2.post.md.txt)
[post 4.md.txt](https://github.com/ObsidianPublisher/obsidian-github-publisher/files/10374191/post.4.md.txt)


**files in github**
https://github.com/peter-hanzo/pub-0/blob/main/docs/second%20my%20post.md
https://github.com/peter-hanzo/pub-0/blob/main/docs/wiki2/post-G3.md


**files in site**
https://lovely-blini-8cc43b.netlify.app/second%20my%20post/
https://lovely-blini-8cc43b.netlify.app/wiki2/post-g3/


1. in githab all is fine, all links work
2. ⚡**in my site once of two internal links not work**⚡
3. the problem change to replace "spaces" with "-" remains in one file

**Comment by Mara-Li** [comment]
As I said before, 
I can't repro this issue.

**Reply by dev-hanzo** [reply]
[deploy log.txt](https://github.com/ObsidianPublisher/obsidian-github-publisher/files/10374300/deploy.log.txt)

**Reply by Mara-Li** [reply]
As I said, 
I don't have this problem on my side. 
What regex do you use for file name. 

![image](https://user-images.githubusercontent.com/30244939/211341457-ccdc6d8c-2890-4ddc-8167-aad6cb981590.png)

It's mine.

**Reply by dev-hanzo** [reply]
![image](https://user-images.githubusercontent.com/117753674/211344786-4ac05265-4d52-4b31-8e81-6dbddf3814b1.png)

**Reply by Mara-Li** [reply]
That's why !
In Javascript, if you don't set any flag in a regex, only the first occurrence will be replaced. 
Write the same regex as mine, and normally it must works!

**Reply by Mara-Li** [reply]
My regex is `/\W+/gi`

**Reply by dev-hanzo** [reply]
**i am sorry**
i thought that allowed use common signs or regex
where can i read more about use regex ?

**Reply by dev-hanzo** [reply]
/\W+/gi
it did not work

![image](https://user-images.githubusercontent.com/117753674/211353184-eaad19e9-1551-4036-bac9-2a66c574f7c5.png)

![image](https://user-images.githubusercontent.com/117753674/211353485-afaf35ee-68b5-47a6-8257-0b272bffd778.png)

**Reply by Mara-Li** [reply]
I really don't understand why it doesn't work. In my side, all my title and filename have their space converted.

**Reply by Mara-Li** [reply]
by the way, I update the Embeded plugin for mkdocs to 2.0.3 to handle properly links in markdown links

**Comment by dev-hanzo** [comment]
**new error** 🔴

![image](https://user-images.githubusercontent.com/117753674/211456888-4261afa3-8da6-453f-93f3-4145ce790785.png)


**sent file**

[last post.md.txt](https://github.com/ObsidianPublisher/obsidian-github-publisher/files/10378720/last.post.md.txt)

**Comment by dev-hanzo** [comment]
> ⚡in my site once of two internal links not work⚡

I still did not can decide this problem

**Comment by Mara-Li** [comment]
.... Internal block doesn't work anywhere outside of obsidian...

**Reply by dev-hanzo** [reply]
that's not the point.
yesterday this link it ok
blocks not work, but link to blocks will mast work (jast it led to file, but not blok)
now this blok deleted, it is not good
see my file please. you will see then first link (block) it is OK, second  link (block) it is not OK

![image](https://user-images.githubusercontent.com/117753674/211495599-7893d9f6-ff7c-4c86-b4eb-ba8bfab2a5d7.png)

**Reply by Mara-Li** [reply]
Fixed in the last update (4.11.7)

**Comment by dev-hanzo** [comment]
Story Finished 👏👏👏. 
I set in YAML "category" in all my files. **After it all work!** 
regex and links work. 
my regex ```/[ ]/gi```
**Only one little thing**: link into Embeded not work, but outside Embeded work



![image](https://user-images.githubusercontent.com/117753674/211633192-b8b7b079-847f-4108-b443-eef66cff7d3c.png)

**Reply by Mara-Li** [reply]
update your requirements to the last version, this bug is now fixed

**Comment by dev-hanzo** [comment]
I see one error ⚠
Internal pictures link work, not like the external pictures link 

![image](https://user-images.githubusercontent.com/117753674/211769170-88f1eb83-2444-4921-b680-9a1852928058.png)

**Comment by Mara-Li** [comment]
Strange bug, thank you, i will fix it asap

**Reply by Mara-Li** [reply]
(every time you read "strange bug" I mean : "I again fucked up with regex")
